packages/ui/src/providers/DocumentInfo/index.tsx

```diff
@@ -260,6 +266,14 @@ const DocumentInfo: React.FC<
     [],
   )
 
+  useEffect(() => {
+    setInitialState(initialState)
```

Comment:
This `useEffect` appears unnecessary. Do you mean to put `initialStateFromProps` as the dependency?

Reply:
In that case, a `useMemo` might make more sense.